staff/staff_views_helper.py

```diff
@@ -208,3 +213,150 @@ def add_volunteers_to_volunteer_dict(classroom):
+
+
+def change_classroom_ta(request):
+    ta_contact = get_contact_by_user_id(request.POST["former_ta"])
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    new_ta = get_contact_by_user_id(request.POST["teacher"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    classroom.teacher_assistant_id = request.POST["teacher"]
+    classroom.save()
+    remove_enrollment(ta_contact, class_offering)
+    ClassEnrollment.objects.get_or_create(
+        created_by=class_offering.created_by,
+        contact=new_ta,
+        status="Enrolled",
+        class_offering=class_offering,
+    )
+
+
+def remove_volunteer(request):
+    vol_contact = get_contact_by_user_id(request.POST["former_vol"])
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    classroom.volunteers.remove(request.POST["former_vol"])
+    remove_enrollment(vol_contact, class_offering)
+
+
+def remove_student(request):
+    student_contact = get_contact_by_user_id(request.POST["former_student"])
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    classroom.students.remove(request.POST["former_student"])
+    remove_enrollment(student_contact, class_offering)
+
+
+def add_volunteers(request):
+    form = AddVolunteersForm(request.POST)
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    if form.is_valid():
+        for volunteer in form.cleaned_data.get("volunteers"):
+            vol_contact = get_contact_by_user_id(volunteer)
+            ClassEnrollment.objects.create(
+                created_by=class_offering.created_by,
+                contact=vol_contact,
+                status="Enrolled",
+                class_offering=class_offering,
+            )
+            classroom.volunteers.add(volunteer)
+
+
+def add_students(request):
+    form = AddStudentsForm(request.POST)
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    if form.is_valid():
+        for student in form.cleaned_data.get("students"):
+            student_contact = get_contact_by_user_id(student)
+            ClassEnrollment.objects.create(
+                created_by=class_offering.created_by,
+                contact=student_contact,
+                status="Enrolled",
+                class_offering=class_offering,
+            )
+            classroom.students.add(student)
+
+
+def remove_enrollment(contact, class_offering):
+    old_enrollment = ClassEnrollment.objects.get(
+        contact=contact, status="Enrolled", class_offering=class_offering
+    )
+    created_by = old_enrollment.created_by
+    old_enrollment.delete()
+    return created_by
+
+
+def get_contact_by_user_id(id):
```

Comment:
Is there a way to do this without two database queries?

Reply:
This raises an important issue which is how we cross-reference users between our postgres database and the salesforce database.  I think there are a couple ways we could approach solving it (I think #2 is what I would favor).  I'll try to think of more as well.  

1.  I could restructure a large portion of the application (which would take a good chunk of time but if you think it is worthwhile I will) to refer to users (pass around users) by their Salesforce id instead of their User id, then this type of double database call wouldn't be necessary.

2.  We could add a field to Contact in the salesforce database that is their user_id in our application and get them that way.

Reply:
Ah sorry, I was reading this as two salesforce queries, not one in postgres and one in salesforce. This is fine.

Probably best for integrity reasons to only have the ids go in one direction.